tools/build_api.py

```diff
@@ -525,6 +527,12 @@ def build_project(src_paths, build_path, target, toolchain_name,
         # Call unified scan_resources
         resources = scan_resources(src_paths, toolchain, inc_dirs=inc_dirs)
 
+        # Generate PSA XML box code from manifests
+        psa_source_pattern = re.compile(r'spm_.+\.[oc]')
+        if any(psa_source_pattern.match(basename(f)) for f in itertools.chain(resources.objects, resources.c_sources)):
```

Comment:
Is there a reason that C++ and  ASM are excluded here?

----

This line is really long, could you split it up?

-----

This seems a bit off that having a psa manifest does not itself trigger the `process_manifest_files`. Why do I need to have a `spm_` prefixed file to cause this to happen? Why not just check for a non-empty `resources.psa_manifests`?

Reply:
this check is due to the nature of the testing compilation of mbed-os
1. due to initialization of SPM in the boot process, we must create the common part of the auto generated code (even if there aren't any manifest present).
2. we don't include ASM and C++ files because our sources are written in c (and we don't see the need to move to C++).
3. We scan the objects to detect SPM sources when compiling tests (with `mbed test`)
4. the scanning of SPM sources is to avoid errors with mbed2 builds (where SPM isn't included)

Reply:
@orenc17A this change may no longer be needed.
SamG suggested `spm` module should move to root folder instead, thus it will always be picked during the compilation (also in mbed2 case) - thus the filtering would no longer be needed.